src/Components/Activities/Detail/ActivityDetail.js

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from 'react';
+//COMPONENTS
+import Title from '../../Title/Title';
+//MUI
+import { CircularProgress, Container, Typography } from '@mui/material';
+//In the future this API call logic should be moved.
+import axios from 'axios';
+const getActivityById = (id) => {
+  return axios.get(`http://ongapi.alkemy.org/api/activities/${id}`);
+};
+
+export default function ActivityDetail({ match }) {
+  const [details, setDetails] = useState({});
+  const { name, description, image } = details;
+  const [success, setSuccess] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const activityId = match.params.id;
+
+  useEffect(() => {
+    getActivityById(activityId)
+      .then((res) => {
+        setSuccess(true);
+        setDetails(res.data.data);
+      })
+      .catch((e) => {
+        setSuccess(false);
+        setDetails(e.response.data);
+      })
+      .finally(() => setIsLoading(false));
+  }, [activityId]);
+
+  if (isLoading) {
+    return (
+      <Container
+        maxWidth="md"
+        sx={{
+          display: 'flex',
+          minHeight: '100vh',
+          justifyContent: 'center',
+          alignItems: 'center',
+        }}>
+        <CircularProgress />
+      </Container>
+    );
+  } else
```

Comment:
this else is not necessary because the return will exit the execution of the code and this block will only be executed if is not loading